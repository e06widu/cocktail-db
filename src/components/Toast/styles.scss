.toast {
    position: fixed;
    top: 100px;
    left: 50%;
    transform: translateX(-50%);
    background-color: #32c20a;
    color: #fff;
    padding: 10px 20px;
    border-radius: 4px;
    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.2);
    z-index: 9999;
}

.toast-message {
    font-size: 16px;
    background: #32c20a;
}