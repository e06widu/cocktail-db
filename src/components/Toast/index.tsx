import React, { useState, useEffect } from 'react';
import './styles.scss'

interface ToastProps {
  message: string;
  duration?: number;
}

const Toast: React.FC<ToastProps> = ({ message, duration = 3000 }) => {
  const [visible, setVisible] = useState(false);

  useEffect(() => {
    setVisible(true);

    const timer = setTimeout(() => {
      setVisible(false);
    }, duration);

    return () => {
      clearTimeout(timer);
    };
  }, [duration]);

  if (!visible) {
    return null;
  }

  return (
    <div className="toast">
      <div className="toast-message">{message}</div>
    </div>
  );
};

export default Toast;
