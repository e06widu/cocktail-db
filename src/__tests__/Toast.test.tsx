import { render } from '@testing-library/react';
import Toast from '../components/Toast';

jest.useFakeTimers();

describe('Toast', () => {
    test('renders toast with message', () => {
      const message = 'Toast message';
  
      const { getByText } = render(<Toast message={message} />);
  
      // Check if the toast message is rendered
      expect(getByText(message)).toBeTruthy();
    });
  
    // test('renders toast for specified duration', () => {
    //   const duration = 3000;
  
    //   const { container } = render(<Toast message="Toast message" duration={duration} />);
  
    //   // Check if the toast is initially visible
    //   expect(container.firstChild).toMatchSnapshot();
  
    //   // Advance timers by the duration
    //   jest.advanceTimersByTime(duration);
  
    //   // Check if the toast is no longer visible after the specified duration
    //   expect(container.firstChild).toMatchSnapshot();
    // });
  });