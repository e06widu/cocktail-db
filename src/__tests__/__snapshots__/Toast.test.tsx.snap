// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast renders toast for specified duration 1`] = `
<div
  class="toast"
>
  <div
    class="toast-message"
  >
    Toast message
  </div>
</div>
`;

exports[`Toast renders toast for specified duration 2`] = `
<div
  class="toast"
>
  <div
    class="toast-message"
  >
    Toast message
  </div>
</div>
`;
